Content.Server/Xenoarchaeology/Equipment/ArtifactAnalyzerSystem.cs

```diff
@@ -31,21 +39,59 @@ private void OnExtractButtonPressed(Entity<AnalysisConsoleComponent> ent, ref An
         if (!_research.TryGetClientServer(ent, out var server, out var serverComponent))
             return;
 
+        // Begin DeltaV - Variables for extraction totals
         var sumResearch = 0;
+        var sumGlimmer = 0;
+        // End DeltaV
+        ArtifactAnalyzerComponent? analyzer = null;
+        if (ent.Comp.AnalyzerEntity is { } analyzerNetEntity)
+        {
+            var analyzerEntityUid = GetEntity(analyzerNetEntity); // Convert NetEntity to EntityUid
+            TryComp<ArtifactAnalyzerComponent>(analyzerEntityUid, out analyzer);
+        }
+
         foreach (var node in _xenoArtifact.GetAllNodes(artifact.Value))
         {
             var research = _xenoArtifact.GetResearchValue(node);
             _xenoArtifact.SetConsumedResearchValue(node, node.Comp.ConsumedResearchValue + research);
+
+            // Begin DeltaV - Only run if we have an artifact ready for extraction
+            if (analyzer != null)
+            {
+                sumGlimmer += (int)(research / (float)analyzer.ExtractRatio);
+                research = (int)(research * GetGlimmerMultiplier(analyzer));
+            }
+            // End DeltaV
             sumResearch += research;
         }
+        if (analyzer != null)
+        {
+            UpdateClientUI(ent, analyzer);
+        }
+
 
-        // 4-16-25: It's a sad day when a scientist makes negative 5k research
```

Comment:
Put this back, please.

I know it's a pointless comment, but the worry is that this comment's in upstream, and git can be really picky during merges. Keeping our diffs/DeltaV-presence as small as possible in upstream files like this one can really help out our upstream mergers by preventing merge conflicts.

In most codebases, little cleanup things like this are usually a good thing. But DeltaV, being a quite hard fork from upstream, lives in a weird coding space where we can't really do that too often.

If you'd like, you can PR cleanups like this in the upstream repo.